writer.go

```diff
@@ -1,130 +1,53 @@
 package protocol
 
 import (
-	"fmt"
 	"time"
 )
 
+type metric struct {
+	name   string
+	tags   []*Tag
+	fields []*Field
+	t      time.Time
+}
+
+func (m *metric) Name() string {
+	return m.name
+}
+func (m *metric) TagList() []*Tag {
+	return m.tags
+}
+
+func (m *metric) FieldList() []*Field {
+	return m.fields
+}
+
+func (m *metric) Time() time.Time {
+	return m.t
+}
+
 // Write writes out data to a line protocol encoder.  Note: it does no sorting.  It assumes you have done your own sorting for tagValues
 func (e *Encoder) Write(name []byte, ts time.Time, tagKeys, tagVals, fieldKeys [][]byte, fieldVals []interface{}) (int, error) {
```

Comment:
This change causes a lot more allocations.  I was hoping to avoid that.

Reply:
I just found out I did this in the wrong place.
I figured I might close this for now and perhaps we can get this fixed in another branch 👍